node/src/actors/chain_manager/mining.rs

```diff
@@ -893,27 +896,26 @@ fn build_block(
             }
         };
 
-        let new_block_weight = block_weight + transaction_weight;
-
-        if new_block_weight <= max_block_weight {
-            value_transfer_txns.push(vt_tx.clone());
-
+        let new_vt_weight = vt_weight.saturating_add(transaction_weight);
+        if new_vt_weight <= max_vt_weight {
             update_utxo_diff(
                 &mut utxo_diff,
                 vt_tx.body.inputs.iter().collect(),
                 vt_tx.body.outputs.iter().collect(),
                 vt_tx.hash(),
             );
+            value_transfer_txns.push(vt_tx.clone());
             transaction_fees += transaction_fee;
-            block_weight += transaction_weight;
+            vt_weight += transaction_weight;
         }
 
-        if new_block_weight == max_block_weight {
+        if new_vt_weight == max_vt_weight {
```

Comment:
If we know the weight of the lighter possible VTT, we can add that to the left operand in the equality and save a crazy amount of iterations here.